### PR TITLE
reside-85: Support for language fallback

### PR DESCRIPTION
**Pull request by richfitz**
Implement fallbacks, so that if a string is not found in one language, something sensible in another is returned.  This will prevent keys from being displayed to the user.

This is required for mrc-1150, in turn required for mrc-787

### CODE REVIEW

File: R/i18n.R [hunk 1]
```diff
@@ -41,18 +41,26 @@
 ##'   method.  Note that the adding a language here does not (yet)
 ##'   mean that failure to load the language is an error.
 ##'
+##' @param fallback The fallback language to use.  The options here
+##'   are to use a character string (a single fallback to use for all
+##'   languages), a character vecto (a series of languages to use in
```
Comment (r-ash) [inline R/i18n.R]: tiny typo - should be "a character vector"

File: tests/testthat/test-i18n.R [hunk 2]
```diff
@@ -172,3 +172,54 @@ test_that("load resource bundles", {
...
+  obj <- i18n(NULL, default_namespace = "common", fallback = "en")
+  obj$add_resource_bundle("en", "common",
+                          traduire_file("examples/structured/en-common.json"))
+  obj$add_resource_bundle("en", "login",
+                          traduire_file("examples/structured/en-login.json"))
+  obj$add_resource_bundle("fr", "common",
+                          traduire_file("examples/structured/fr-common.json"))
+  expect_equal(obj$t("hello", language = "en"), "hello world")
+  expect_equal(obj$t("hello", language = "fr"), "salut le monde")
+  expect_equal(obj$t("hello", language = "ko"), "hello world")
```
Comment (r-ash) [inline tests/testthat/test-i18n.R:187]: The fallback covers missing keys too?

Worth adding a test to show how that behaves?

**Review by r-ash — CHANGES_REQUESTED** [review]
Question about whether the fallback covers missing keys in languages and how that is handled. And how is it handled if the key is missing from the fallback too? Perhaps that is already tested somewhere else though.

File: R/i18n.R (hunk 1 above)
Comment (richfitz) [reply R/i18n.R]: thanks :)

File: tests/testthat/test-i18n.R (hunk 2 above)
Comment (richfitz) [reply tests/testthat/test-i18n.R:187]: done

**Pull request merged**